### PR TITLE
Feat: Allow query params to return all values

### PR DESCRIPTION
**Pull request by rgoupil**
## Description
It is currently impossible to read all the values passed for a query param when it is present more than once, only the last one is kept.

For instance when for the following query `...?param=1&param=2`:  
`@Query('param')` would return `"2"`  
`@Query()` would return `{ "param": "2" }`  

This PR introduce a `@QueryAll` decorator that is similar `@Query` except it always return all the values instead of only the last one in the URL. Using the same example as earlier:  
`@QueryAll('param')` would return `["1", "2"]`  
`@QueryAll()` would return `{ "param": ["1", "2"] }`

I also slightly changed the way `@Param()` get the path parameters (it's untyped but I swear that's how you're supposed to do it) in order to not return query params as well (which could create some conflict issues).
While technically a breaking change, using `@Param()` to return the query params is most likely a terrible idea that I have never seen in any other frameworks.

---
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

---
# Checklist:
- [x] I have run `deno lint` AND `deno fmt` AND `deno task test` and got no errors.
- [x] I have followed the contributing guidelines of this project as mentioned in [CONTRIBUTING.md](/CONTRIBUTING.md)
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/Savory/Danet/pulls) for the same update/change?
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes needed to the documentation

### CODE REVIEW

**Comment by Sorikairox** [comment]
`Query` and `QueryAll` are redundant IMO. There is no valid use-case where getting the last one with `Query` instead of using `QueryAll` is valid.

`QueryAll` should be renamed `Query` and if it takes no parameters it returns the same thing as `Query` without parameters.


`Param` does exist 😛 Check out https://docs.nestjs.com/controllers#full-resource-sample example 😄

File: src/router/controller/params/decorators.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { getQuery } from '../../../deps.ts';
+import { getQuery } from 'https://deno.land/x/oak@v10.5.1/helpers.ts';
```
[the remark this answers is not in the document]
Comment (rgoupil) [reply src/router/controller/params/decorators.ts:1]: Damn auto-import, nice catch!

**Comment by rgoupil** [comment]
> `Query` and `QueryAll` are redundant IMO. There is no valid use-case where getting the last one with `Query` instead of using `QueryAll` is valid.
> 
> `QueryAll` should be renamed `Query` and if it takes no parameters it returns the same thing as `Query` without parameters.
> 
> `Param` does exist 😛 Check out https://docs.nestjs.com/controllers#full-resource-sample example 😄

I know `Param` exists 😛 but in the current state it returns the exact same object than `Query`, which is not right since `Query` is for query parameters and `@Param` is for path parameters!
For reference, Nest gives these values:
@Query(key?: string) | req.query / req.query[key]
@Param(key?: string) | req.params / req.params[key]

You can see that `Query` and `Param` are not supposed to return the same value 😉

**Comment by rgoupil** [comment]
> `Query` and `QueryAll` are redundant IMO. There is no valid use-case where getting the last one with `Query` instead of using `QueryAll` is valid.
> 
> `QueryAll` should be renamed `Query` and if it takes no parameters it returns the same thing as `Query` without parameters.

I'll try to make a better job at explaining what QueryAll does, although it probably show that we should change it's name.
Using `...?test=toto&myparam=1&myparam=2` as an example URL:
* `@Query('myparam')` return the last value for the query param "myparam": `"2"`
* `@QueryAll('myparam')` return all the values for the query param "myparam": `["1", "2"]`
* `@Query()` return an object containing the last value of all query parameters: `{ "test": "toto", "myparam": "2" }`
* `@QueryAll()` return an object containing all the values of all query parameters: `{ "test": ["toto"], "myparam": ["1", "2"] }`

Please note that the behaviour of `Query` was not changed in any other way than not returning the path params anymore (which was a bug too 😋).

**Review by Sorikairox — APPROVED** [review]
LGTM :shipit:

**Pull request merged**